runtime/bin/directory_linux.cc

```diff
@@ -277,28 +274,27 @@ static bool DeleteRecursively(PathBuffer* path) {
 
   // Iterate the directory and delete all files and directories.
   int path_length = path->length();
-  dirent entry;
-  dirent* result;
-  while (NO_RETRY_EXPECTED(readdir_r(dir_pointer, &entry, &result)) == 0) {
+  while (true) {
+    dirent* result = readdir(dir_pointer);
     if (result == NULL) {
```

Comment:
If `readdir` returns `NULL` it might be end-of-stream **or** error. Why not implement the same error handling as above?